packages/ui/src/ui/pages/query-tracker/module/vcsNavigation/actions.ts

```diff
@@ -0,0 +1,259 @@
+import {ThunkAction} from 'redux-thunk';
+import {RootState} from '../../../../store/reducers';
+import axios from 'axios';
+import {Action} from 'redux';
+import {wrapApiPromiseByToaster} from '../../../../utils/utils';
+import {
+    DirectoryItem,
+    FileItem,
+    Repositories,
+    setBranch,
+    setBranches,
+    setList,
+    setPath,
+    setPreview,
+    setRepositories,
+    setRepository,
+    setVcs,
+    setVcsConfig,
+} from './vcsNavigationSlice';
+import {selectVcsConfig, selectVcsNavigation} from './selectors';
+import {getQueryFiles} from '../query/selectors';
+import {updateQueryDraft} from '../query/actions';
+import guid from '../../../../common/hammer/guid';
+import {selectFileEditor} from '../queryFilesForm/selectors';
+import {setFileEditor} from '../queryFilesForm/queryFilesFormSlice';
+import {GithubRepository, GitlabRepository, VcsConfig} from '../../../../../shared/vcs';
```

Comment:
Probably it is better to declare some type in `shared/vcs` with `GithubRepository | GitlabRepository`